core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -581,6 +640,18 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    */
   final def interruptible: ZIO[R, E, A] = interruptStatus(InterruptStatus.Interruptible)
 
+  /**
+   * Returns an effect that when interrupted returns immediately.
+   * However, the effect and its interruption occurs in a forked fiber
+   */
+  final def interruptibleFork: ZIO[R, E, A] =
+    ZIO.uninterruptible {
+      for {
+        fiber <- self.interruptible.fork.daemon
+        res   <- fiber.join.interruptible.onInterrupt(fiber.interrupt.fork.daemon)
```

Comment:
Do you need to fork the interruption? Also, it may be necessary to get the fiber id first, and then use `interruptAs(parentId)`

Reply:
Does the interruption not need to be forked so that the parent fiber's interruption completes immediately?

Reply:
Yes, it is necessary to fork the interruption as per @dariusrobson initial attempt.

Reply:
Ah, good idea to use `interruptAs`. We want to push the parent fiber id into the children interruptions.